lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp

```diff
@@ -181,40 +181,39 @@ status_code_e S1apStateManager::read_ue_state_from_db() {
 }
 
 void S1apStateManager::create_s1ap_imsi_map() {
-  s1ap_imsi_map_ = new s1ap_imsi_map_t();
+  proto_map_uint32_uint64_t imsi_map;
+  s1ap_imsi_map_ = new oai::S1apImsiMap();
+  if (!s1ap_imsi_map_) {
+    OAILOG_ERROR(LOG_S1AP, "Failed to allocate memory for s1ap_imsi_map_");
+    return;
+  }
```

Comment:
The nullptr check is not needed because in C++ new throws std::bad_alloc by default.